### PR TITLE
pythonPackages.magic-wormhole: 0.10.5 -> 0.11.2

### PR DESCRIPTION
**Pull request by marsam**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by worldofpeace — CHANGES_REQUESTED** [review]
The `txaio` update is already in [staging-next](https://github.com/NixOS/nixpkgs/blob/staging-next/pkgs/development/python-modules/txaio/default.nix)

**Comment by marsam** [comment]
@worldofpeace do I change the base branch to `staging` or wait until `staging` is merged into master?

**Comment by worldofpeace** [comment]
> @worldofpeace do I change the base branch to `staging` or wait until `staging` is merged into master?

Currently magic-wormhole is broken on master because a failure in the dependency chain.
That fix is going to be delivered to master from `staging-next`. (txaio update)

So merging this without your `txaio` bump is good for master.

**Comment by marsam** [comment]
okay I've removed the txaio bump

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build pythonPackages.magic-wormhole python3Packages.magic-wormhole

File: pkgs/development/python-modules/magic-wormhole/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/magic-wormhole/default.nix]: Why are these tests being removed?

Comment (marsam) [reply pkgs/development/python-modules/magic-wormhole/default.nix]: these fail in sandboxed build because they require networking

Comment (worldofpeace) [reply pkgs/development/python-modules/magic-wormhole/default.nix]: I'm using sandboxing and it was good without this...

File: pkgs/development/python-modules/magic-wormhole/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/magic-wormhole/default.nix]: This test had to have been fine if we're correcting it.

File: pkgs/development/python-modules/magic-wormhole/default.nix
Comment (marsam) [reply pkgs/development/python-modules/magic-wormhole/default.nix]: sorry for the delay, building it with `nix-build -I . -A pythonPackages.magic-wormhole --option sandbox true` gives me the following errors  https://gist.github.com/c9d700fb7ac00e4ab303adee1c497921
fwiw right now I'm building it on darwin and linux (remote-builder)

File: pkgs/development/python-modules/magic-wormhole/default.nix
Comment (marsam) [reply pkgs/development/python-modules/magic-wormhole/default.nix]: sorry I missed that, I've updated the PR with a patch fixing the tests. Please let me know what you think

File: pkgs/development/python-modules/magic-wormhole/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/magic-wormhole/default.nix]: I think we still need this https://github.com/warner/magic-wormhole/blob/13102d3eeded33a4ad749078de9bbca5000ffbe6/src/wormhole/test/test_cli.py#L280

Comment (worldofpeace) [inline pkgs/development/python-modules/magic-wormhole/default.nix]: I think we still need this as the point `running the system's locale binary instead of the one from Nix` should still stand right?

But a patch with `substituteAll` would be better.

**Comment by worldofpeace** [comment]
I glanced at the tests and they seem to only want networking with loopback, which is the only networking allowed within a sandboxed build. (with yourself)

cc @Mic92

File: pkgs/development/python-modules/magic-wormhole/default.nix
Comment (marsam) [reply pkgs/development/python-modules/magic-wormhole/default.nix]: not anymore we pass `glibcLocales` as buildInput and patch to `getProcessOutputAndValue` os.environ to honor it. I've checked that the tests affected are executed correctly

**Comment by marsam** [comment]
Reading a few threads, seems to be a problem with sandbox builds on darwin, even the darwin builders don't run with sandboxing enabled anymore. Tested on darwin (without sandbox) and Linux (with sandbox).
Rebased and I believe is ready to be merged.

File: pkgs/development/python-modules/magic-wormhole/default.nix
Comment (worldofpeace) [reply pkgs/development/python-modules/magic-wormhole/default.nix]: Thanks :+1:

**Comment by worldofpeace** [comment]
How convenient that this finally hit also #49742.

I see you rebased with that so I can actually check if this builds here FTR.

@GrahamcOfBorg build pythonPackages.magic-wormhole python3Packages.magic-wormhole

Other than that I think https://github.com/NixOS/nixpkgs/pull/50634/commits/61139609037861418946245ceb4efb3dbf317eb4 could be fixup'd into https://github.com/NixOS/nixpkgs/pull/50634/commits/007d186584af81ae0ff14b271aca1147a7f01f95.

**Comment by marsam** [comment]
> Other than that I think 6113960 could be fixup'd into 007d186.

done

**Review by worldofpeace — APPROVED** [review]
Tested execution and everything still works :+1: 

cc @c0bw3b

**Comment by worldofpeace** [comment]
@marsam 
Thank you for your contribution and co-operation :sparkle:

**Pull request merged**
